src/components/Dialog/DialogsManager.ts

```diff
@@ -0,0 +1,178 @@
+type DialogId = string;
+
+export type GetOrCreateParams = {
+  id: DialogId;
+  isOpen?: boolean;
+};
+
+export type Dialog = {
+  close: () => void;
+  id: DialogId;
+  isOpen: boolean | undefined;
+  open: (zIndex?: number) => void;
+  remove: () => void;
+  toggle: () => void;
+  toggleSingle: () => void;
+};
+
+type DialogEvent = { type: 'close' | 'open' | 'openCountChange' };
+
+const dialogsManagerEvents = ['openCountChange'] as const;
+type DialogsManagerEvent = { type: typeof dialogsManagerEvents[number] };
+
+type DialogEventHandler = (dialog: Dialog) => void;
+type DialogsManagerEventHandler = (dialogsManager: DialogsManager) => void;
+
+type DialogInitOptions = {
+  id?: string;
+};
+
+const noop = (): void => undefined;
+
+export class DialogsManager {
+  id: string;
+  openDialogCount = 0;
+  dialogs: Record<DialogId, Dialog> = {};
+  private dialogEventListeners: Record<
+    DialogId,
+    Partial<Record<DialogEvent['type'], DialogEventHandler[]>>
+  > = {};
+  private dialogsManagerEventListeners: Record<
+    DialogsManagerEvent['type'],
+    DialogsManagerEventHandler[]
+  > = { openCountChange: [] };
```

Comment:
Just a suggestion: to avoid _yet another event listener_ maybe you can reuse our `StateStore` here? I think this will work nicely with dialog ids as selectors etc.

Reply:
Will take a look into that.